contracts/ibc2/src/contract.rs

```diff
@@ -9,13 +9,16 @@ use serde::{Deserialize, Serialize};
 #[derive(Default, Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
 pub struct State {
     ibc2_packet_receive_counter: u32,
+    ibc2_timeout_counter: u32,
 }
 
 #[cw_serde]
 #[derive(QueryResponses)]
 pub enum QueryMsg {
     #[returns(State)]
     QueryState {},
+    #[returns(u32)]
+    QueryTimeoutCounter {},
```

Comment:
There isn't really a need for this query, since the `QueryState` already contains the timeout counter.